theano/sandbox/cuda/opt.py

```diff
@@ -314,7 +314,7 @@ def local_gpu_split(node):
             split_node = host_input.owner
             new_op = GpuSplit(split_node.op.len_splits)
             return [new_op(gpu_from_host(split_node.inputs[0]),
-                           *split_node.inputs[1:])[host_input.index]]
+                           *split_node.inputs[1:])]
```

Comment:
There is not direct test of that case. It is hard do that that. The old code was blatently wrong. It always returneded a subtensor of the answer we needed. So it was alwayrs returning an error. I tested the case that give me this problem and the error message disapeared.